docs/developer_overview.md

```diff
@@ -88,8 +87,7 @@ $ bazel run //iree/tools:iree-opt -- \
 
 ### iree-translate
 
-The `iree-translate` program translates from a .mlir input file into an IREE
-module.
+`iree-translate` converts MLIR input into external formats like IREE modules.
```

Comment:
Maybe good to reference and mlir-translate and https://mlir.llvm.org/getting_started/Glossary/#translation here. And could also reference https://mlir.llvm.org/getting_started/Glossary/#conversion above regarding `iree-opt`

Reply:
Good ideas, added more context links.